Fundamentals/Setup/Server-Setup/Load-Balancing/azure-web-apps-v9.md

```diff
@@ -0,0 +1,81 @@
+
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "Hosting": {
+                "LocalTempStorageLocation" : "EnvironmentTemp"
+            }
+        }
+    }
+}
+```
+
+## AppDomain synchronization
+
+Each application runs inside an [AppDomain](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/application-domains) which is like a subprocess within the web app process. When an ASP.Net application restarts, the current AppDomain 'winds down' while another AppDomain is started; meaning there can be more than 1 live AppDomain during a restart. Restarts can occur in many scenarios including when an Azure Web App auto transitions between hosts, you scale the instances or you utilise slot swapping.
+
+Several file system based services in Umbraco such as the Published Cache and Lucene files can only be accessed by a single AppDomain at once. Umbraco manages this synchronization by an object called `IMainDom`. By default this uses a system-wide locking mechanism but this default mechanism doesn't work in Azure Web Apps so we need to swap it out for an alternative database locking mechanism by using the following appSetting:
+
+```json
+{
+    "Umbraco": {
+        "CMS": {
+            "Global": {
+                "MainDomLock" : "SqlMainDomLock"
+            }
+        }
+    }
+}
+```
+
+Apply this setting to both the __SCHEDULINGPUBLISHER__ Administrative server and the __SUBSCRIBER__ scalable public-facing servers.
+
+## Steps to set-up an environment
+
+1. Create an Azure SQL database
+2. Install Umbraco on your backoffice administrative environment and ensure to use your Azure SQL Database
+3. Install Umbraco on your scalable public facing environment and ensure to use your Azure SQL Database
+4. Test: Perform some content updates on the administrative environment, ensure they work successfully on that environment, then verify that those changes appear on the scalable public facing environment
+5. Fix the backoffice environment to be the SCHEDULINGPUBLISHER scheduling server and the scalable public facing environment to be SUBSCRIBERs - see [Setting Explicit Server Roles](flexible-advanced.md#explicit-schedulingpublisher-server)
+
+:::note
+Ensure all Azure resources are located in the same region to avoid connection lag
+:::
+
+## Scaling
+
+**Do not scale your backoffice administrative environment** this is not supported and can cause issues.
+
+The public facing subscriber Azure Web Apps can be manually or automatically scaled up or down and is supported by Umbraco's load balancing.
+
+## Deployment considerations
+
+Since you have 2 x web apps, when you deploy you will need to deploy to both places - There are various automation techniques you can use to simplify the process. That is outside the scope of this article.
```

Comment:
```suggestion
Since you have 2 x web apps, when you deploy you will need to deploy to both places - There are automation techniques you can use to simplify the process. That is outside the scope of this article.
```